PlastomeRegionBurstAndAlign.py

```diff
@@ -340,165 +322,154 @@ def main(args):
-                with open(out_fn_aligned_nucl, 'w') as hndl:
-                    hndl.write(stdout)
-        else:
-            log.critical("No items in nucleotide main dictionary to process")
-            raise Exception()
-    # --------------------------------------------------------------------------#
-    # CONDUCT PROTEIN ALIGNMENT AND BACK-TRANSLATION TO NUCLEOTIDES
-    if select_mode == 'cds':
-        action = "conducting multiple sequence alignment based on protein sequence data, followed by a back-translation to nucleotides"
-        log.info("%s" % action)
-        ###
-        try:
-            proteinalign_and_backtranslate(main_d_prot, out_dir, log)
-        except:
-            raise Exception()
-    # --------------------------------------------------------------------------#
-    # CONVERT FASTA ALIGNMENT TO NEXUS ALIGNMENT AND APPEND TO LIST
+def conduct_protein_alignment_and_back_translation(main_d_prot, out_dir, log):
+    action = "conducting multiple sequence alignment based on protein sequence data, followed by back-translation to nucleotides"
+    log.info("%s" % action)
+    try:
+        proteinalign_and_backtranslate(main_d_prot, out_dir, log)
+    except:
+        raise Exception()
+
+def convert_fasta_alignment_to_nexus(main_d_nucl, out_dir):
     action = "converting FASTA alignment to NEXUS alignment"
     log.info("%s" % action)
-    ###
     successList = []
+
     for k in main_d_nucl.keys():
         aligned_nucl_fasta = os.path.join(out_dir, 'nucl_' + k + '.aligned.fasta')
         aligned_nucl_nexus = os.path.join(out_dir, 'nucl_' + k + '.aligned.nexus')
-        # Convert FASTA alignment to NEXUS alignment
         try:
-            Bio.AlignIO.convert(aligned_nucl_fasta, 'fasta', aligned_nucl_nexus, 'nexus', molecule_type='DNA')
+            AlignIO.convert(aligned_nucl_fasta, 'fasta', aligned_nucl_nexus, 'nexus', molecule_type='DNA')
         except:
             log.warning("Unable to convert alignment of `%s` from FASTA to NEXUS" % k)
-            # raise Exception()
-            continue  # skip to next k in loop, so that k is not included in successList
-        # Import NEXUS and append to list for concatenation
+            continue
         try:
-            # ipdb.set_trace()
-            alignm_nexus = Bio.AlignIO.read(aligned_nucl_nexus, 'nexus')
+            alignm_nexus = AlignIO.read(aligned_nucl_nexus, 'nexus')
             hndl = io.StringIO()
-            Bio.AlignIO.write(alignm_nexus, hndl, 'nexus')
+            AlignIO.write(alignm_nexus, hndl, 'nexus')
             nexus_string = hndl.getvalue()
-            nexus_string = nexus_string.replace('\n' + k + '_',
-                                                '\nconcatenated_')  # IMPORTANT: Stripping the gene name from the sequence name
-            alignm_nexus = Bio.Nexus.Nexus.Nexus(nexus_string)
-            successList.append((k, alignm_nexus))  # Function 'Bio.Nexus.Nexus.combine' needs a tuple.
```

Comment:
Please keep comments such as this in the code while we develop it.

    # Function 'Bio.Nexus.Nexus.combine' needs a tuple.